docs/topics/arrays.md

```diff
@@ -320,28 +294,34 @@ fun main() {
 ### Convert to Map
 
 To convert an array to a `map`, use the [`toMap()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-map.html) function.
-Only an array of [`Pair`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/) can be converted to a map: `Array<Pair<K,V>>`.
-The first value of a `Pair` instance becomes a key and the second becomes a value.
+Only an array of [`Pair<K,V>`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/) can be converted to a map.
+The first value of a `Pair` instance becomes a key and the second becomes a value. In the example below, we use the [infix notation](functions.md#infix-notation) to call the [`to()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/to.html) function to create tuples of `Pair`.
 
 ```kotlin
 fun main() {
 //sampleStart
-    val pairArray = arrayOf(Pair("apple","120"), Pair("banana","150"), Pair("cherry","90"), Pair("apple","140"))
+    val pairArray = arrayOf("apple" to 120, "banana" to 150, "cherry" to 90, "apple" to 140)
 
     // Convert to Map {apple=140, banana=150, cherry=90}
     // The keys are fruits and the values are their number of calories
     // Note how keys must be unique, so the latest value of "apple"
     // overwrites the first
     println(pairArray.toMap())
-    
+
 //sampleEnd
 }
 ```
 
-## Primitive type arrays
+> To convert primitive type arrays to object type arrays, use [`toTypedArray()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-typed-array.html).
```

Comment:
Why is this warning now above the introduction of primitive type arrays?

Reply:
Human error. Fixed now!